### ISSUE TITLE
undefined error fetching localhost:3000/render/http://localhost:1337/?wc-inject-shadydom

### ISSUE THREAD

**Issue by hekod777** [GoogleChrome/rendertron#81]
after watching the polymer summit video I decided to give rendertron a try

1. I run a local rendertron at localhost: 3000

2. I run a simple expressJS + polymer 2.0 app at localhost:1337

3. I enter localhost:1337 in the input field in the rendertron UI. it worked very well. it flatten shadow dom. and render the page correctly except the words in <p> tags which it just ignored.

4.  then I added the middleware below

```
app.use(rendertron.makeMiddleware({
  proxyUrl: 'localhost:3000/render/',
  injectShadyDom: true,
}));
```
to `sever.js`

5. the middleware can actually detect if a bot is making a request which is good

6. but it also gives me this error when a bot is making a request (yes I modified the middleware a bit to see what might go wrong)
`[rendertron middleware] undefined error fetching localhost:3000/render/http://localhost:1337/?wc-inject-shadydom=true`

7. I simply copy and pasty 
`localhost:3000/render/http://localhost:1337/?wc-inject-shadydom=true`
to a browser. and I can actually see a rendered version of the html by rendertron.

I'd like to know what actually went wrong and threw that error. and how to fix it.

thx!

**Comment by samuelli** [comment]
cc @aomarks 

Thanks for the report. Does it take a while to render the page? It may be hitting a timeout, or there could be a different bug with the middleware.

Do you have anything that could help us reproduce this? eg. a polymer 2 app that you reproduce this with?

**Comment by hekod777** [comment]
this error pops up as soon as the bot request is made. it probably has nothing to do with the 10 sec rendering time. 

the code below is in server.js

```

const express = require ('express');
const app = express();
const bodyParser = require ('body-parser');
const path = require('path');
const rendertron = require('rendertron-middleware');

let port = process.env.PORT || 1337;

app.use(bodyParser.urlencoded({extended: true}));

app.use('/bower_components', express.static(path.join(__dirname, 'bower_components')));

app.use('/src', express.static(path.join(__dirname, 'src')));

app.use(rendertron.makeMiddleware({
  proxyUrl: 'localhost:3000/render',
  injectShadyDom: true,
}));

app.use(express.static('files'));

app.get('/', function(req,res){
  res.sendFile(path.resolve(__dirname, 'index.html'));
});

app.listen(port, function(){
  console.log('listening to ' + port);
});
```

and the index.html and the file structure is the same as 
`polymer init `
then select polymer 2 application

--------------

I also tested it by deploying the app on heroku and using `https://render-tron.appspot.com/render/` as end point.

when a bot request is made, it has this `<my-app></my-app> ` in the body tag
but there is simply nothing inside my-app tag, not even a shadown dom.

but `https://render-tron.appspot.com/render/` can actually render my heroku page correctly.

**Comment by samuelli** [comment]
@hekod777: it looks like `http://` is missing from `proxyUrl` in that code sample. Can you try it with the full URL?

**Comment by hekod777** [comment]
@samuelli 

Thank you for the advise. now the error is gone. and after adding some console.logs in rendertron. I can see that if a bot sends a request to my app. it will hit the rendertron/render end point.

and rendertron actually successfully rendered the page with a status code of either 200 or 304.

but whats inside `<my-app></my-app>` is still empty.

it seems that there are some problems with serving the rendered html to the bot?

**Comment by hekod777** [comment]
interesting

after adding the following line to the rendertron-middleware everything worked!

`renderUrl += '=true';`

**Comment by frankiefu** [comment]
I saw the same problem too. Here is the PR: https://github.com/GoogleChrome/rendertron/pull/82

**Comment by samuelli** [comment]
Thanks @hekod777, @frankiefu, let me know if there's further issues.

**Comment by samuelli** [comment]
The fix will be released in middleware 0.1.2. #83
